src/middleware/ReasonRouter.ts

```diff
@@ -0,0 +1,51 @@
+import { Session } from 'ch-node-session-handler';
+import { NextFunction, Request, Response } from 'express';
+import { provide } from 'inversify-binding-decorators';
+import { BaseMiddleware } from 'inversify-express-utils';
+
+import { loggerInstance } from 'app/middleware/Logger';
+import { ApplicationData, APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { ReasonType } from 'app/models/fields/ReasonType';
+import { APPLICATION_DATA_UNDEFINED, SESSION_NOT_FOUND_ERROR } from 'app/utils/CommonErrors';
+import { Feature } from 'app/utils/Feature';
+import { isFeatureEnabled } from 'app/utils/FeatureChecker';
+import { ILL_PERSON_PAGE_URI, OTHER_REASON_DISCLAIMER_PAGE_URI } from 'app/utils/Paths';
+
+@provide(ReasonRouter)
+export class ReasonRouter extends BaseMiddleware {
+
+    public handler(req: Request, res: Response, next: NextFunction): void {
+
+        if (!isFeatureEnabled(Feature.ILLNESS_REASON)) {
+            return next();
+        }
+
+        const session: Session | undefined = req.session;
+        if (!session) {
+            throw SESSION_NOT_FOUND_ERROR;
+        }
+
+        const applicationData: ApplicationData | undefined = session.getExtraData(APPLICATION_DATA_KEY);
+        if (!applicationData) {
+            throw APPLICATION_DATA_UNDEFINED;
+        }
+
+        const reason = applicationData.appeal.currentReasonType;
+        let redirect: string | undefined;
+
+        switch(reason) {
+            case ReasonType.illness:
```

Comment:
can we not just add the routing logic to where we're building the navigation control in the `ChooseAppealReasonController.ts`? It no longer makes sense to navigate to the OtherReasonDisclaimer page by default

Reply:
I agree - sending the user to the Other Reason journey by default doesn't sit well with me - but I could work out how to do this within the navigation definition. I couldn't work out how to gain access to the session (or better yet, the form data, so that we don't need this new session variable) from there. Is this something we can do??

Reply:
Resolved as suggested.